File: quickstart/java-spring-ai/src/main/java/com/example/ai/controller/AiController.java
package com.example.ai.controller;

import com.example.ai.service.AiService;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/ai")
public class AiController {

    private final AiService aiService;

    public AiController(AiService aiService) {
        this.aiService = aiService;
    }

    @PostMapping("/generate")
    public ResponseEntity<AiResponse> generate(@Valid @RequestBody AiRequest request) {
        AiResponse response = aiService.generateText(request);
        return ResponseEntity.ok(response);
    }

    public record AiRequest(
        @NotBlank(message = "Prompt cannot be blank")
        String prompt,
        @Positive(message = "Max tokens must be positive")
        int maxTokens
    ) {}

    public record AiResponse(
        String generatedText,
        int tokensUsed,
        double confidence
    ) {}
}

File: quickstart/dotnet-ai/src/AiApi/Controllers/AiController.cs
using Microsoft.Extensions.AI;

namespace AiApi.Controllers;

[ApiController]
[Route("api/[controller]")]
public class AiController : ControllerBase
{
    private readonly AiService _aiService;

    public AiController(AiService aiService)
    {
        _aiService = aiService;
    }

    [HttpPost("generate")]
    public async Task<IActionResult> Generate([FromBody] AiRequest request)
    {
        if (!ModelState.IsValid)
        {
            return BadRequest(ModelState);
        }

        var response = await _aiService.GenerateTextAsync(request);
        return Ok(response);
    }
}

public record AiRequest(
    [Required] [StringLength(1000, MinimumLength = 1)] string Prompt,
    [Range(1, 1000)] int MaxTokens = 100
);

public record AiResponse(
    string GeneratedText,
    int TokensUsed,
    double Confidence
);
